profiles/common/modules/custom/nexteuropa_piwik/nexteuropa_piwik.install

```diff
@@ -91,4 +107,67 @@ function nexteuropa_piwik_enable() {
   );
   // Grant the permissions to PIWIK administrator role.
   user_role_change_permissions($piwik_admin_role->rid, $piwik_admin_permissions);
+
+  _nexteuropa_piwik_grant_permission_for_admin();
+}
+
+/**
+ * Returns the schema array.
+ */
+function _nexteuropa_piwik_get_schema_array() {
+  return array(
+    'description' => 'Next Europa PIWIK rules',
+    'fields' => array(
+      'id' => array(
+        'type' => 'serial',
+        'not null' => TRUE,
+        'description' => 'Primary Key: Unique PIWIK rule ID.',
+      ),
+      'rule_language' => array(
+        'type' => 'varchar',
+        'length' => 12,
+        'not null' => TRUE,
+        'description' => 'The language for which a given rule applies.',
+      ),
+      'rule_path' => array(
+        'type' => 'text',
+        'description' => 'Path where a given rule applies.',
+      ),
+      'rule_path_type' => array(
+        'type' => 'varchar',
+        'length' => 16,
+        'not null' => TRUE,
+        'description' => 'The rule path type.',
+      ),
+      'rule_section' => array(
+        'type' => 'varchar',
+        'length' => 128,
+        'not null' => TRUE,
+        'description' => 'The name of the site section where a given rule applies.',
+      ),
+    ),
+    'primary key' => array('id'),
+  );
+}
+
+/**
+ * Implements hook_schema().
+ */
+function nexteuropa_piwik_schema() {
+  $schema = array();
+  $schema['nexteuropa_piwik_rule'] = _nexteuropa_piwik_get_schema_array();
+
+  return $schema;
+}
+
+/**
+ * Add the PIWIK rules table.
+ */
+function nexteuropa__update_7100() {
```

Comment:
I partially agree. I think that it should create a schema during the installation (nexteuropa_piwik is enabled by default in the build/install). Doing that in separate function can have an impact on uninstall process where if hook schema is defined in .install file the cleaning operation is performed automatically. So to recap I can use 
```
$schema = module_invoke('my_module', 'schema');
db_create_table('table_name', $schema['table_name']);
```
to create schema instead of using hook_update but I would like to keep the hook_schema definition in the .install file. Is it ok for you @Fefaine?